FromDoppler/doppler-webapp
# Fallback to Doppler Legacy public pages based on configuration

Hi team, 

It allows us to disable our Login, Signup and ForgotPassword pages and redirect them to Doppler Legacy pages.

I resurrected a previous automated test to cover most of the new code.

---

You can verify how is it working **based on build environment**:

* QA (**should redirect** to legacy pages)
    * Login <https://cdn.fromdoppler.com/doppler-webapp/qa-build327/#/login>
    * Signup <https://cdn.fromdoppler.com/doppler-webapp/qa-build327/#/signup>
    * Forgot Passwprd <https://cdn.fromdoppler.com/doppler-webapp/qa-build327/#/forgot-password>
* INT (**should NOT redirect** to legacy pages)
    * Login <https://cdn.fromdoppler.com/doppler-webapp/int-build326/#/login>
    * Signup <https://cdn.fromdoppler.com/doppler-webapp/int-build326/#/signup>
    * Forgot Passwprd <https://cdn.fromdoppler.com/doppler-webapp/int-build326/#/forgot-password>

Or **forcing the behavior using URL parameters**:

* QA (**forced to NOT redirect** to legacy pages)
    * Login <https://cdn.fromdoppler.com/doppler-webapp/qa-build327/#/login?force-webapp>
    * Signup <https://cdn.fromdoppler.com/doppler-webapp/qa-build327/#/signup?force-webapp>
    * Forgot Passwprd <https://cdn.fromdoppler.com/doppler-webapp/qa-build327/#/forgot-password?force-webapp>
* INT (**forced to redirect** to legacy pages)
    * Login <https://cdn.fromdoppler.com/doppler-webapp/int-build327/#/login?force-legacy>
    * Signup <https://cdn.fromdoppler.com/doppler-webapp/int-build327/#/signup?force-legacy>
    * Forgot Passwprd <https://cdn.fromdoppler.com/doppler-webapp/int-build327/#/forgot-password?force-legacy>
